### ISSUE TITLE
[ERROR i18n_embed::requester] Unable to parse your locale when $LANG is not set

### ISSUE THREAD

**Issue by gendx** [kellpossible/cargo-i18n#49]
As reported in a project that depends on `i18n_embed` (https://github.com/str4d/rage/issues/156), running `DesktopLanguageRequester::requested_languages()` in a minimalist Docker environment (on Linux) where `$LANG` is not set causes the following error message to be logged.

```
[ERROR i18n_embed::requester] Unable to parse your locale: ParserError(InvalidLanguage)
```

This is in particular visible for all dependent programs of `i18n_embed` that use the `env_logger` mechanism for example.

However, this error message is somewhat misleading because it isn't actually fatal to the program (if I understood correctly, `requested_languages()` would simply return an empty `Vec`). I have various ideas of how this could be improved.

- Could `DesktopLanguageRequester` be extended with some option parameter (e.g. in a `new_with_options` constructor) to explicitly choose to gracefully fallback (i.e. without error message) if no locale could be found/parsed?
- Could the logged error message be more specific about which language was invalid and couldn't be parsed, and/or specify the source where a language was searched (e.g. which environment variable was read)?
- Could the case where `$LANG` is unset be handled in a more graceful manner - given that it's not really a "parsing" error from an "invalid language", and that this variable cannot always be expected to be set?

What do you think?

**Comment by kellpossible** [comment]
Hi @gendx thanks for reporting! I'll take a look at this over the next few days and hopefully get back to you on it soon

**Comment by kellpossible** [comment]
It's more of a problem with upstream https://github.com/rust-locale/locale_config/issues/6, but for now I created https://github.com/kellpossible/cargo-i18n/pull/50 to hopefully work around the issue, and report the potential issue of the empty tag using info::debug instead of info::error

**Comment by gendx** [comment]
Thanks for taking a look!
It seems that we're getting to some solution by jumping one level deeper in the dependencies.

And regardless of the upstream fix, I think that #50 would be an improvement by clarifying what caused the error.
